### PR TITLE
fix: Trim trailing slash from Gitea URL

### LINKED ISSUE #3487: Gitea Publish URL contains an extra /
### What happened?

When following [the documentation](https://goreleaser.com/scm/gitea/#urls), publishing a release to Gitea creates a URL that Gitea cannot parse properly.

This incorrect URL will be published when pushing to a homebrew repository, where trying to install will 404.

### How can we reproduce this?

https://try.gitea.io/goreleaserbugtester9000/GoreleaserTest

```yaml
gitea_urls:
  api: https://try.gitea.io/api/v1/
```

creates an (incorrect) publishing URL at https://try.gitea.io//goreleaserbugtester9000/GoreleaserTest/releases/tag/v0.0.4
Homebrew repo: https://try.gitea.io/goreleaserbugtester9000/homebrew/src/branch/main/GoreleaserTest.rb#L12

while

```yaml
gitea_urls:
  api: https://try.gitea.io/api/v1
```

creates a (correct) publishing URL at https://try.gitea.io/goreleaserbugtester9000/GoreleaserTest/releases/tag/v0.0.5
Homebrew repo: https://try.gitea.io/goreleaserbugtester9000/homebrew/src/branch/main/TestCorrectURL.rb#L12

### goreleaser version

```bash
goreleaser version 1.12.3
commit: e27e6a9e8c66ec5d2bbcafb1c047068bcf250269
built at: 2022-10-20T12:30:56Z
built by: goreleaser
goos: linux
goarch: amd64
module version: v1.12.3, checksum: h1:sTJXdkGQO9eAP6pwEUazaD6ERJewZ7yzQexWEGPUVg4=

https://goreleaser.com
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

Related Gitea issue: https://github.com/go-gitea/gitea/issues/20462

If this behaviour is intentional, I can make a PR to fix the docs.

### PR DESCRIPTION
**Pull request by SamTherapy**
This PR will truncate any trailing slash left by specifying only a Gitea API URL.

Fixes #3487

I have not added any tests for this.

### CODE REVIEW

**Comment by SamTherapy** [comment]
Strange, this did not fail when I tested it locally.

File: internal/pipe/defaults/defaults.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ func (Pipe) Run(ctx *context.Context) error {
 		}
 
 		ctx.Config.GiteaURLs.Download = strings.ReplaceAll(apiURL, "/api/v1", "")
+		ctx.Config.GiteaURLs.Download = strings.TrimSuffix(apiURL, "/")
```
Comment (caarlos0) [inline internal/pipe/defaults/defaults.go:39]: ah, I think this should be

```suggestion
		ctx.Config.GiteaURLs.Download = strings.TrimSuffix(ctx.Config.GiteaURLs.Download, "/")
```

Comment (caarlos0) [inline internal/pipe/defaults/defaults.go]: or this entire thing could be:

```suggestion
		ctx.Config.GiteaURLs.Download = strings.TrimSuffix(strings.ReplaceAll(apiURL, "/api/v1", ""))
```

Comment (SamTherapy) [reply internal/pipe/defaults/defaults.go]: I like this better. It makes it impossible to make the mistake I just did. Thank you!

**Pull request merged**